src/systems/audio-system.js

```diff
@@ -176,26 +186,71 @@ export class AudioSystem {
     }
   }
 
-  addAudio(mixerTrack, audioNode) {
-    audioNode.gain.connect(this.mixer[mixerTrack]);
+  addAudio({ sourceType, node }) {
+    let outputNode = node;
+    if (isThreeAudio(node)) {
+      outputNode = node.gain;
+    }
+    outputNode.connect(this.mixer[sourceType]);
   }
 
-  removeAudio(audioNode) {
-    audioNode.gain.disconnect();
-    audioNode.sourceType !== "empty" && audioNode.disconnect();
+  removeAudio({ node }) {
+    let outputNode = node;
+    if (isThreeAudio(node)) {
+      outputNode = node.gain;
+    }
+    outputNode.disconnect();
   }
 
   updatePrefs() {
-    const { globalVoiceVolume, globalMediaVolume } = window.APP.store.state.preferences;
+    const { globalVoiceVolume, globalMediaVolume, globalSFXVolume } = window.APP.store.state.preferences;
     let newGain = (globalMediaVolume !== undefined ? globalMediaVolume : 100) / 100;
     this.mixer[SourceType.MEDIA_VIDEO].gain.setTargetAtTime(newGain, this.audioContext.currentTime, GAIN_TIME_CONST);
 
+    newGain = (globalSFXVolume !== undefined ? globalSFXVolume : 100) / 100;
+    this.mixer[SourceType.SFX].gain.setTargetAtTime(newGain, this.audioContext.currentTime, GAIN_TIME_CONST);
+
     newGain = (globalVoiceVolume !== undefined ? globalVoiceVolume : 100) / 100;
     this.mixer[SourceType.AVATAR_AUDIO_SOURCE].gain.setTargetAtTime(
       newGain,
       this.audioContext.currentTime,
       GAIN_TIME_CONST
     );
+
+    if (MediaDevicesManager.isAudioOutputSelectEnabled && window.APP.mediaDevicesManager) {
+      const sinkId = window.APP.mediaDevicesManager.selectedSpeakersDeviceId;
+      const isDefault = sinkId === DEFAULT_DEVICE_ID;
+      const sink = isDefault ? this._sceneEl.audioListener.getInput() : this.audioDestination;
+      this.mixer[SourceType.AVATAR_AUDIO_SOURCE].disconnect();
+      this.mixer[SourceType.AVATAR_AUDIO_SOURCE].connect(sink);
+      this.mixer[SourceType.MEDIA_VIDEO].disconnect();
+      this.mixer[SourceType.MEDIA_VIDEO].connect(sink);
+      this.mixer[SourceType.SFX].disconnect();
+      this.mixer[SourceType.SFX].connect(sink);
+      if (isDefault) {
+        if (this.outputMediaAudio) {
+          this.outputMediaAudio.pause();
+          this.outputMediaAudio.srcObject = null;
+          this.outputMediaAudio = null;
+        }
+      } else {
+        // Swithing the audio sync is only supported in Chrome at the time of writing this.
+        // It also seems to have some limitations and it only works on audio elements. We are piping all our media through the Audio Context
```

Comment:
I haven't found any incompatibilities between both but I guess the best way to do it test it in a group with different setups. I'll deploy it and we can test in a meeting.

Yes, it's only applied in case you enable it otherwise it won't use the hack.